e2e/e2e_test.go

```diff
@@ -87,6 +92,85 @@ var _ = Describe("Whereabouts functionality", func() {
 				Expect(inRange(ipv4TestRange, secondaryIfaceIP)).To(Succeed())
 			})
 		})
+
+		Context("stateful set tests", func() {
+			const (
+				initialReplicaNumber = 20
+				ipPoolNamespace      = "kube-system"
+				namespace            = "default"
+				serviceName          = "web"
+				selector             = "app=" + serviceName
+				statefulSetName      = "statefulthingy"
+			)
+
+			BeforeEach(func() {
+				var err error
+				_, err = clientInfo.provisionStatefulSet(statefulSetName, namespace, serviceName, initialReplicaNumber)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(
+					clientInfo.Client.CoreV1().Pods(namespace).List(
+						context.TODO(), metav1.ListOptions{LabelSelector: selector})).To(
+					WithTransform(func(podList *core.PodList) int { return len(podList.Items) }, Equal(initialReplicaNumber)))
+			})
+
+			AfterEach(func() {
+				const noReplicas = 0
```

Comment:
This is more feedback for myself...that I want to run by you. I like this tight scoping for consts, i.e. allow const to exist at the minimal possible scope. I think I'm going to follow this for my comprehensive e2e PR.

Example: `singlePodName   = "whereabouts-basic-test"` declared somewhere around line 30-40 should be removed from being a global const, and should instead live under the `BeforeEach()` of "Single pod tests" context, since it's only ever used there.

And if a const ends up being reused its scope can be changed as needed.

Reply:
Yes, that's the idea. 

If the value of the const happens to be the same ... It might make sense to move it to a common place, but, that's not a silver bullet.